### DISCUSSION TITLE
Add audio pin?

### DISCUSSION THREAD

**Discussion by teoAlivanoglou** [tshino/softcam#58] [Ideas]
Any plans to support audio? If not, any hints on where I should get started to try to implement it myself?

**Comment by tshino** [comment]
No plans.
As far as I know, audio and video are entirely separate concepts in DirectShow.
That means making a virtual audio input device (virtual microphone) might require something like developing an audio version of Softcam from scratch.

**Comment by teoAlivanoglou** [comment]
According to [Microsoft documentation](https://learn.microsoft.com/en-us/windows/win32/directshow/csource), it seems a CSource can contain multiple streams:


> The CSource class is a base class for implementing source filters. A filter derived from CSource contains one or more output pins derived from the [CSourceStream](https://learn.microsoft.com/en-us/windows/win32/directshow/csourcestream) class. Each output pin creates a worker thread that pushes media samples downstream.


Now, I might be completely misunderstanding how the whole thing works. I won't lie, I dislike the whole windows API.

**Comment by tshino** [comment]
I thought this issue has been finished. However your comment might be useful information for someone else, so I decided to convert this issue to a discussion thread.
